lib/ticks-to-tree.js

```diff
@@ -99,71 +113,187 @@ function ticksToTree (ticks, options = {}) {
+      }
+      if (sourceMaps[fileName]) {
+        sourceMapFiles.unshift(sourceMaps[fileName])
+      }
+    }
+
+    if (existsSync(fileName)) {
+      // read last 2 lines to accomodate for a line-break
+      const lastLine = await readLastLines.read(fileName, 2)
+      const sourceMapMatch = lastLine.match(/\/\/[#@]\s+sourceMappingURL=([^\s]+)/i)
+      if (sourceMapMatch) {
+        sourceMapFiles.unshift(resolve(dirname(fileName), sourceMapMatch[1]))
+      }
+    }
+
+    for (let i = 0; i < sourceMapFiles.length; i++) {
+      if (sourceMapFiles[i]) {
+        // convert relative source map paths to absolute paths
+        if (!isAbsolute(sourceMapFiles[i])) {
+          sourceMapFiles[i] = resolve(dirname(fileName), sourceMapFiles[i])
+        }
+
+        if (existsSync(sourceMapFiles[i])) {
+          try {
+            debug(`loading source map from ${sourceMapFiles[i]}`)
+            const sourceMap = JSON.parse(readFileSync(sourceMapFiles[i]))
+            loadedSourceMaps[fileName] = await new SourceMapConsumer(sourceMap)
+            if (!loadedSourceMaps[fileName].sourceRoot) {
+              loadedSourceMaps[fileName].sourceRoot = dirname(sourceMapFiles[i])
+            }
+            break
+          } catch (e) {
+            console.error(e)
+          }
+        }
+      }
+    }
+
+    loadedSourceMaps[fileName] = loadedSourceMaps[fileName] || false
+    return loadedSourceMaps
+  }
+}
+
+function nameFromSourceMap (name, loadedSourceMaps) {
+  const key = name.split(' ', 2)
+  const [ fileName, line, column ] = key[1].split(':')
 
-  function addToUnmergedTree (stack) {
-    var lastFrame = null
-    stack.forEach((frame, ix) => {
-      // TODO need an elegant way to place inlined frames into *other* stacks
-      // where the function has disappeared – for now we just remove inlined
-      // stacks from merged graphs
-      // adding children here means we avoid unnecessary arrays
-      if (ix > 0) lastFrame.children = lastFrame.children || []
-      const children = (ix === 0) ? unmerged.children : lastFrame.children
-      const child = children.find(({ fn, S }) => {
-        return fn === frame.name && S === frame.S
+  if (loadedSourceMaps[fileName]) {
+    try {
+      const source = loadedSourceMaps[fileName].originalPositionFor({
+        line: parseInt(line),
+        column: parseInt(column)
       })
 
-      if (child === undefined) {
-        frame.fn = frame.name
-        if (frame.S === 1) frame.name = '~' + frame.name
-        else if (frame.S === 2) frame.name = '*' + frame.name
-        else if (frame.S === 3) frame.name = '~' + frame.name + ' [INLINABLE]'
-        else if (frame.S === 4) frame.name = '*' + frame.name + ' [INLINABLE]'
-        children.push(frame)
-      } else frame = child
+      if (source.source) {
+        name = `${key[0]} ${source.source}:${source.line}:${source.column}`
+      }
+    } catch (err) {
+      console.error(err, fileName, line, column)
```

Comment:
Are we sure this is correct? If we can't read the file, is it safe to just carry on?

Reply:
Is it actually safe?